### PR TITLE
Release/phase 3.0

### PR DESCRIPTION
**Pull request by Blaarkies**

### CODE REVIEW

File: game-engine/default-config.json [hunk 1]
```diff
@@ -40,11 +58,14 @@
...
     "doNothing": 0,
     "invalidCommand": -4
   },
   "wormSelectTokens": {
     "count": 5
   },
-  "csvSeparator": ","
+  "csvSeparator": ",",
+  "battleRoyaleStart": 0.25,
+  "battleRoyaleEnd": 0.875
```
Comment (DaanKeun) [inline game-engine/default-config.json]: Shouldn't this rather be specified in rounds?

File: game-engine/game-rules.md [hunk 2]
```diff
@@ -176,5 +209,20 @@ The total score value is determined by adding together the player's average worm
...
+The worms' disagreements here have initiated geological activity. A slow flood of lava will engulf the entire map if they cannot sort out their differences
+* From round **100** lava will slowly creep onto the edges of the map
+* At round **350** lava will have filled the entire map except for a small circular region in the center of the map
+* Lava will replace most cells as it floods over the map
+  * Air cells are replaced by lava cells
+  * Dirt cells are not replaced
+  * Deep space cells are not replaced
+  * When a digging worm forms new air cell in the flooded region, it is replaced by lava at the start of the next round
+    * The same happens for destroyed dirt cells via a Banana Bomb
+* Any worm standing on top of a lava cell, will sustain **3** damage every round
```
Comment (DaanKeun) [inline game-engine/game-rules.md]: '... every round that it remains in the lava'

File: game-engine/default-config.json (hunk 1 above)
Comment (DaanKeun) [inline game-engine/default-config.json]: Shouldn't this be specified in rounds rather than percentages?

File: game-engine/state-files.md [hunk 3]
```diff
@@ -25,20 +26,25 @@
...
+      * count → *The amount of Snowballs in this worm's inventory*
+      * freezeRadius → *The radius to which targets will be frozen*
       
       
 * opponents → *The player details of your opponents. An example of an opponent looks like the following:*
   * id → *The Id number of this player*
   * score → *This player's score points currently acquired*
   * currentWormId  → *The Id number of the opponent player's worm that is currently in play*
-  * previousCommand → *The command which this opponent bot supplied to the engine on the last round as a string.  e.g. 'move 20 9'.  Note that it doesn't contain 'C;\<number\>;'*
+  * previousCommand → *The command which this player supplied to the engine on the last round as a string.  e.g. 'move 20 9'.  Note that it doesn't contain 'C;\<number\>;'*
```
Comment (DaanKeun) [inline game-engine/state-files.md]: shouldn't this refer to the opponent?

File: game-engine/default-config.json (hunk 1 above)
Comment (Blaarkies) [reply game-engine/default-config.json]: That would make these dependent on the maxRounds number (if these are bigger than, they won't do anything then...i think)

File: game-engine/state-files.md (hunk 3 above)
Comment (Blaarkies) [reply game-engine/state-files.md]: It does, but to the player object here. The wording opponent does make it more clear though 🤔

**Pull request merged**